test/end-to-end/offer.js

```diff
@@ -298,6 +299,139 @@ describe("Offer endpoint tests", () => {
             });
         });
 
+        describe("Before reaching the offers limit while having past offers", () => {
+            const testOffers = [];
+            for (let i = 1; i < CompanyConstants.offers.max_number; ++i) {
+                testOffers.push({
+                    title: `Active Test Offer no${i}`,
+                    publishDate: new Date(Date.now() - (DAY_TO_MS)),
+                    publishEndDate: new Date(Date.now() + (DAY_TO_MS)),
+                    description: "For Testing Purposes",
+                    contacts: ["geral@niaefeup.pt", "229417766"],
+                    jobType: "SUMMER INTERNSHIP",
+                    fields: ["DEVOPS", "MACHINE LEARNING", "OTHER"],
+                    technologies: ["React", "CSS"],
+                    // owner: will be set in beforeAll
+                    // ownerName: will be set in beforeAll
+                    location: "New Testing Avenue, New Testing, 234",
+                });
+            }
+
+            testOffers.push({
+                title: "Past Test Offer",
+                publishDate: new Date(Date.now() - (2 * (DAY_TO_MS))),
+                publishEndDate: new Date(Date.now() - (DAY_TO_MS)),
+                description: "For Testing Purposes",
+                contacts: ["geral@niaefeup.pt", "229417766"],
+                jobType: "SUMMER INTERNSHIP",
+                fields: ["DEVOPS", "MACHINE LEARNING", "OTHER"],
+                technologies: ["React", "CSS"],
+                // owner: will be set in beforeAll
+                // ownerName: will be set in beforeAll
+                location: "New Testing Avenue, New Testing, 234",
+            });
+
+            beforeAll(async () => {
+                await Offer.deleteMany({});
+
+                testOffers.forEach((offer) => {
+                    offer.owner = test_company._id;
+                    offer.ownerName = test_company.name;
+                });
+
+                await Offer.create(testOffers);
+            });
+
+            afterAll(async () => {
+                await Offer.deleteMany({});
+            });
+
```

Comment:
Add a test for future publishDates (existing future offers and candidate offer also in the future, something like this:
```
This should pass:

now
|---------------------|           Offer 1            | -------------
|-------------------------|           Offer 2                |-----------
|----------------------------------------| Candidate Offer |-------
|-------------------------|           Offer 4                |-----------
|--|    Offer 5    |------------------------------------------------
```

```
This should not:

now
|---------------------|           Offer 1            | -------------
|-------------------------|           Offer 2                |-----------
|----------------------------------------| Candidate Offer |-------
|-------------------------|           Offer 4                |-----------
|------------------------------------------|    Offer 5    |------------
```

Note: By doing these diagrams I see the problem of checking the dates is not trivial, for example this should be allowed but it won't according to the restrictions I asked for:

```
Notice how at each moment there are only 4 active offers
now
|---------------------|           Offer 1    | -------------
|-------------------------|           Offer 2                |-----------
|----------------------------------------| Candidate Offer |-------
|-------------------------|           Offer 4                |-----------
|-------------------------------------------------|    Offer 5    |------------
```

For now, I ask you not to worry about this, as it is a use case that I doubt will have any meaningful impact in the near future, however we should create an issue regarding this.

Reply:
In this regard, in order to check the dates properly in situations like the second example, I added one condition to the ones you provided in the first comment:


>* `publishEndDate` between `publishDate` and `publishEndDate` of candidate Offer
>* `publishStartDate` between publishDate` and `publishEndDate` of candidate Offer

* `publishStartDate` before `publishDate` of candidate Offer and `publishEndDate` after `publishEndDate` of candidate Offer.

Reply:
Nice find! That was indeed missing...

Reply:
I pushed the commit with the new changes. As you said, I left that special case for a future issue